multiversx_sdk/network_providers/account_awaiter_test.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+
+from multiversx_sdk.core.address import Address
+from multiversx_sdk.core.transaction import Transaction
+from multiversx_sdk.core.transaction_computer import TransactionComputer
+from multiversx_sdk.network_providers.account_awaiter import AccountAwaiter
+from multiversx_sdk.network_providers.api_network_provider import \
+    ApiNetworkProvider
+from multiversx_sdk.network_providers.resources import AccountOnNetwork
+from multiversx_sdk.testutils.mock_network_provider import (
+    MockNetworkProvider, TimelinePointMarkCompleted, TimelinePointWait)
+from multiversx_sdk.testutils.utils import create_account_egld_balance
+from multiversx_sdk.testutils.wallets import load_wallets
+
+
+class TestTransactionAwaiter:
+    provider = MockNetworkProvider()
+    watcher = AccountAwaiter(
+        fetcher=provider,
+        polling_interval_in_milliseconds=42,
+        timeout_interval_in_milliseconds=42 * 42,
+        patience_time_in_milliseconds=0
+    )
+
+    @pytest.mark.only
+    def test_await_on_balance_increase(self):
+        alice = Address.new_from_bech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th")
+        initial_balance = self.provider.get_account(alice).balance
+
+        # adds 7 EGLD to the account balance
+        self.provider.mock_account_balance_timeline_by_address(
+            alice,
+            [TimelinePointWait(40), TimelinePointWait(40), TimelinePointWait(45), TimelinePointMarkCompleted()]
+        )
+
+        def condition(account: AccountOnNetwork):
+            return account.balance == initial_balance + create_account_egld_balance(7)
+
+        account = self.watcher.await_on_condition(alice, condition)
+        assert account.balance == create_account_egld_balance(1007)
+
+    @pytest.mark.networkInteraction
+    def test_on_network(self):
```

Comment:
Maybe say something about the balance increase in the test name? Like the one above, the one with the mock.

Reply:
renamed to `test_await_for_account_balance_increase_on_network`.